### ISSUE TITLE
QueueingBasicConsumer is obsolete

### ISSUE THREAD

**Issue by dennyferra** [HangfireIO/Hangfire.SqlServer.RabbitMq#4]
'QueueingBasicConsumer' is obsolete: 'Deprecated. Use EventingBasicConsumer or a different consumer interface implementation instead'

https://github.com/rabbitmq/rabbitmq-tutorials/issues/58

**Comment by nithrous** [comment]
Is EventingBasicConsumer better to use in hangfire queue provider?

**Comment by dennyferra** [comment]
@nithrous Yes, there are a few benefits of using EventingBasicConsumer, basically: 

`QueingBasicConsumer` does not auto-recover and it was a work around for a message dispatcher issue which no longer exists (in Rabbit MQ). Additionally performance can be improved because `EventingBasicConsumer` doesn't use any queueing mechanism whereas `QueueingBasicConsumer` uses `SharedQueue<T>` which is basically .NET Queue with locks.

I should be able to make these changes within the next few weeks. You can also submit a PR if you wish to make the changes before I get to it.

**Comment by dennyferra** [comment]
Just referencing pull request: #7

**Comment by hin-longkid** [comment]
@dennyferra I wonder when will this PR be merge to master?

**Comment by dennyferra** [comment]
@hin-longkid Merged

**Comment by dennyferra** [comment]
Had to revert, see https://github.com/HangfireIO/Hangfire.SqlServer.RabbitMq/pull/9 for details.

This week is busy so I will look into the issues in more detail on Friday (March 3) and should have a new version with the necessary changes.
